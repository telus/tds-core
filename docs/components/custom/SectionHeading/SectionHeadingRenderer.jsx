import React from 'react'
import PropTypes from 'prop-types'

import Heading from '../../../../packages/Heading/Heading'
import FlexGrid from '../../../../packages/FlexGrid/FlexGrid'
import Box from '../../../../packages/Box/Box'

import styles from './SectionHeadingRenderer.modules.scss'

const SectionHeadingRenderer = ({ id, depth, toolbar, children }) => {
  const level = Math.min(depth, 4)

  return (
    <Box below={3}>
      <Heading level={`h${level}`} id={id}>
        <FlexGrid limitWidth={false} gutter={false}>
          <FlexGrid.Row>
            <FlexGrid.Col>
              <Box inline between="space-between">
                {children}
                <div className={styles.toolbar}>{toolbar}</div>
              </Box>
            </FlexGrid.Col>
          </FlexGrid.Row>
        </FlexGrid>
      </Heading>
    </Box>
  )
}

SectionHeadingRenderer.propTypes = {
  children: PropTypes.node.isRequired,
  toolbar: PropTypes.node.isRequired,
  id: PropTypes.string.isRequired,
  depth: PropTypes.number.isRequired,
}

export default SectionHeadingRenderer
