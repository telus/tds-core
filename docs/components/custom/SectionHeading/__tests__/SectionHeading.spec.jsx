import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import SectionHeadingRenderer from '../SectionHeadingRenderer'

describe('SectionHeadingRenderer', () => {
  it('renders', () => {
    const Toolbar = () => <div>The toolbar</div>
    const sectionHeading = shallow(
      <SectionHeadingRenderer id="a-section" depth={2} toolbar={<Toolbar />}>
        A Section
      </SectionHeadingRenderer>
    )

    expect(toJson(sectionHeading)).toMatchSnapshot()
  })
})
