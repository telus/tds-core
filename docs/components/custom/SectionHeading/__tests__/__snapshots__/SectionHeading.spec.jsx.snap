// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeadingRenderer renders 1`] = `
<Box
  below={3}
  inline={false}
  tag="div"
>
  <Heading
    id="a-section"
    invert={false}
    level="h2"
  >
    <Flexbox
      direction="row"
    >
      A Section
      <div
        className="toolbar"
      >
        <Toolbar />
      </div>
    </Flexbox>
  </Heading>
</Box>
`;
